pettingzoo/utils/wrappers/order_enforcing.py

```
@@ -134,13 +107,25 @@ def __iter__(self) -> AECOrderEnforcingIterator[AgentID, ObsType, ActionType]:
 
 
 class AECOrderEnforcingIterator(AECIterator[AgentID, ObsType, ActionType]):
-    def __next__(self) -> AgentID:
-        agent = super().__next__()
+    def __init__(
+        self, env: OrderEnforcingWrapper[AgentID, ObsType, ActionType], max_iter: int
+    ):
         assert hasattr(
-            self.env, "_has_updated"
+            env, "_has_updated"
         ), "env must be wrapped by OrderEnforcingWrapper"
+        # this is set during the super call to init, so setting it here
+        # is redundant. However, it silences pyright errors because it tells
+        # pyright that self.env is an OrderEnforcingWrapper (which may not be
+        # strictly true, but it should have OrderEnforcingWrapper somewhere
+        # in the wrapper list). This might be better handled by Protocols,
+        # but this approach works.
+        self.env = env  # silence pyright errors
+        super().__init__(env, max_iter)
+
+    def __next__(self) -> AgentID:
+        agent = super().__next__()
         assert (
-            self.env._has_updated  # pyright: ignore[reportGeneralTypeIssues]
+            self.env._has_updated
```

I think we can just do ignore general type issues at the top of the file instead of these ugly in line ones (assuming that’s the main reason you changed this)

---

I re-added this per above comment. I think the end of line one is better in this case, even though it is ugly. Ignoring any error that might show up in the file isn't great. This is a generic catch-all error too, so it has a bigger chance of hiding something important that we don't know about.